Python 3: Port from commands module to utils.process

hsmj1412 opened the pull request:
Signed-off-by: Junxiang Li <junli@redhat.com>

hsmj1412 commented:
@clebergnu Sorry, this line is copied from commands source code, has been fixed

czhang0 commented:
#2510

apahim commented on avocado/utils/process.py:
```diff
@@ -1349,3 +1349,39 @@ def system_output(cmd, timeout=None, verbose=True, ignore_status=False,
+    sts = result.exit_status
+    if text[-1:] == '\n':
+        text = text[:-1]
+    return sts, text
```
We have no such function returning this exact information - (exit_code, combined_output), but we do have that information, meaning you can use what we already have in place:
```python
>>> print(process.run('ls .git foo', allow_output_check='combined', ignore_status=True).exit_status)
2

>>> print(process.run('ls .git foo', allow_output_check='combined', ignore_status=True).stdout)
ls: cannot access 'foo': No such file or directory
.git:
branches
COMMIT_EDITMSG
config
FETCH_HEAD
GIT_COLA_MSG
HEAD
hooks
index
info
logs
objects
ORIG_HEAD
packed-refs
refs
``` 

Would that be enough? If not, you can use the `allow_output_check='combined'` here instead of `cmd += ' 2>&1'` and also accept all the others `run()` arguments in `getstatusoutput()`, setting your defaults accordingly  (`timeout=None`, `verbose=False`, `ignore_status=True`, `allow_output_check='combined'`, `shell=True`, ...).

apahim commented on avocado/utils/process.py:
```diff
@@ -1349,3 +1349,39 @@ def system_output(cmd, timeout=None, verbose=True, ignore_status=False,
+
+def getoutput(cmd):
+    """
+    Becauce commands module is removed in Python3 and it redirect stderr
```
s/Becauce/Because/

apahim commented on avocado/utils/process.py:
```diff
@@ -1349,3 +1349,39 @@ def system_output(cmd, timeout=None, verbose=True, ignore_status=False,
+
+def getstatusoutput(cmd):
+    """
+    Becauce commands module is removed in Python3 and it redirect stderr
```
s/Becauce/Because/

apahim commented on avocado/utils/process.py:
```diff
@@ -1349,3 +1349,39 @@ def system_output(cmd, timeout=None, verbose=True, ignore_status=False,
+    :param cmd: Command line to run.
+    :type cmd: str
+
+    :return: Exit status and command output(stdout or stderr).
```
output(stdout **and** stderr)

hsmj1412 commented:
@apahim Thank you for your help!!! I have fixed the problem you pointed out. And I noticed the doc in process.run is old, there was not 'combined' in allow_output_check doc, so I updated the doc. Finally, I am puzzled that the 'none' in allow_output_check is not used, it will be replaced to 'both'?

apahim commented on avocado/utils/process.py:
```diff
@@ -1349,3 +1376,40 @@ def system_output(cmd, timeout=None, verbose=True, ignore_status=False,
+    return getstatusoutput(cmd)[1]
+
+
+def getstatusoutput(cmd):
```
I meant:

```python
def getstatusoutput(cmd, timeout=None, verbose=False, ignore_status=True,
                    allow_output_check='combined', shell=True, env=None,
                    sudo=False, ignore_bg_processes=False)
    ...
    result = run(cmd, timeout, verbose, ignore_status,
                allow_output_check, shell, env,
                sudo, ignore_bg_processes)
    ...
```

The same applies to `getoutput()`.

apahim commented:
@hsmj1412 `allow_output_check='none'` will disable the `sdtout`/`stderr` drainers to files, but the `CmdResult` class will still carry the `sdtout`/`stderr` content.

hsmj1412 commented:
@apahim Updated. Thx

clebergnu commented:
@hsmj1412 it'd be helpful (and it's part of Avocado's contribution guidelines/workflow) to send a v2, detailing the changes when compared to v1.  I'm closing this PR so that you can send a v2.
